### ISSUE TITLE
.env variables take precedence over the outer environment variables

### ISSUE THREAD

**Issue by mvasin** [bkeepers/dotenv#319]
### Steps to reproduce

`.env` file:
```
RACK_ENV=test
DB_NAME=database_${RACK_ENV}
```

`test.rb` file:
```
def display_envs()
  puts
  puts "RACK_ENV=#{ENV['RACK_ENV']}"
  puts "DB_NAME=#{ENV['DB_NAME']}"
  puts
end

puts "Before loading .env..."
display_envs

require 'dotenv'
Dotenv.load

puts "After loading .env..."
display_envs
```

Output:
```
$ RACK_ENV=development ruby test.rb
Before loading .env...

RACK_ENV=development
DB_NAME=

After loading .env...

RACK_ENV=development
DB_NAME=database_test
```
### Expected behavior
`RACK_ENV` from `.env` should not leak in when there is `RACK_ENV` in the outer environment already set.
After loading `.env` file I expected DB_NAME to be `database_development`.

As readme states,
>  it won't overwrite existing environment variables as dotenv assumes the deployment environment has more knowledge about configuration than the application does.

### Actual behavior
In this particular situation a variable already defined in the outer environment does get overwritten.

### System configuration
**dotenv version**: 2.2.1
**Ruby version**: 2.4.2

**Comment by cbjoseph** [comment]
I opened #323 to address this.

**Comment by jonmagic** [comment]
Closes by #323
